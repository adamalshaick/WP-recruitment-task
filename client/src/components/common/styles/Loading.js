import styled, { keyframes } from "styled-components";

const load = keyframes`
0% {
    opacity: 0;
  }
  50% {
    opacity: 0.5;
  }
  100% {
    opacity: 0;
  }
`;

export const Dots = styled.div`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translateX(-50%);
  font-size: 4rem;
  opacity: 0.5;
  animation: ${load} 1.5s infinite;
`;
