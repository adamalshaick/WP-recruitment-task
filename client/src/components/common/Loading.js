import React from "react";
import { Dots } from "./styles/Loading";

export const Loading = () => {
  return <Dots>...</Dots>;
};

export default Loading;
