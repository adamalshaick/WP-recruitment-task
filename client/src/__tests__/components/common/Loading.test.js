import React from "react";
import Loading from "../../../components/common/Loading";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<Loading />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
