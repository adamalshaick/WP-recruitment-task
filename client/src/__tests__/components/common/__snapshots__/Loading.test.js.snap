// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<styled.div>
  ...
</styled.div>
`;
